src/processor/nonce.service.ts

```diff
@@ -23,12 +23,30 @@ export class NonceService implements OnApplicationBootstrap {
       numberOfKeys: RedisUtils.NUMBER_OF_NONCE_KEYS_TO_CHECK,
       lua: fs.readFileSync('lua/incrementNonce.lua', 'utf8'),
     });
+    redis.defineCommand('peekNonce', {
+      numberOfKeys: RedisUtils.NUMBER_OF_NONCE_KEYS_TO_CHECK,
+      lua: fs.readFileSync('lua/peekNonce.lua', 'utf-8'),
+    });
   }
 
   async onApplicationBootstrap() {
     this.accountId = createKeys(this.configService.getProviderAccountSeedPhrase()).publicKey;
-    const nextNonce = await this.getNextNonce();
-    this.logger.log(`nonce is set to ${nextNonce}`);
+    const nextNonce = await this.peekNextNonce();
+    this.logger.log(`Initialized nonce to ${nextNonce}`);
+  }
+
+  async peekNextNonce(): Promise<number> {
```

Comment:
So this is just used for logging in `onApplicationBootstrap` or it's been used in other places?

Reply:
Yeah, that's all. I added it while debugging because previously it was calling `getNextNonce` which was causing a bug. I decided to leave it in.